### PR TITLE
docs: update documentation for all apps

### LINKED ISSUE #337: [Derisk] Update documentation for all apps
1. Before applying, Read our [Contributor Guidelines](https://github.com/CarmineOptions/derisk-research/blob/master/CONTRIBUTING.md) and [README](https://github.com/CarmineOptions/derisk-research/blob/master/README.md). 
2. In [`README.m`](https://github.com/CarmineOptions/derisk-research/blob/master/README.md) update documentation per apps:
- `dashboard_app`
- `web_app` 
- `data_handler`
3. This documentation should have anchor to another REAMDE.md in theirs folders
4. Also, add some chapter, how to run in docker each app in main `REAMDME.md`
5. If you can see that you can improve documentation in apps, please do it

### PR DESCRIPTION
**Pull request by Luluameh**
closes #337 

## Description
- update the main [README.md](https://github.com/CarmineOptions/derisk-research/blob/master/README.md) file at the root of the project

## Comment
- I noticed that some directory don't a readme file but i still referenced them
     - [shared](https://github.com/CarmineOptions/derisk-research/tree/master/apps/shared)
     - [dashboard_app](https://github.com/CarmineOptions/derisk-research/tree/master/apps/dashboard_app)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,39 +2,89 @@
...
+- [`web_app`](./apps/web_app/README.md) - Main web application interface
+- [`legacy_app`](./apps/legacy_app/README.md) - Legacy application functionality
+- [`dashboard_app`](./apps/dashboard_app/README.md) - Analytics dashboard
+- [`shared`](./apps/shared/README.md) - Common code shared between the components
+
+## Quick Start Guide
+
+### Prerequisites
+- Docker installed on your machine (v19.03+ recommended).
+- Docker Compose installed (v1.27+ recommended).
```
Comment (djeck1432) [inline README.md:15]: v2 would be better, some of contributors had issue with v1.~

**Review by djeck1432 — CHANGES_REQUESTED** [review]
Could you please also update structure in other documents like REAMDME.md for data_handler and web_app
also I can't see anything about `dashboard_app`

**Comment by djeck1432** [comment]
@Luluameh Any updates?

**Comment by Luluameh** [comment]
hello @djeck1432 I  am on it and will soon make a push

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**